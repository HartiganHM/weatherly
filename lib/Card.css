.card {
	flex: 1;
	border: 0.01rem solid ghostwhite;
	margin: 1rem;
}
