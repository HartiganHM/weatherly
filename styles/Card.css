.card {
  flex: 1;
  border: 0.01rem solid ghostwhite;
  margin: 1rem;
  padding: 1rem;
}

.temp {
  margin: 0;
}
