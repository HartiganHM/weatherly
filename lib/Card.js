import React from 'react'
import '../styles/Card.css'

const Card = (props) => {
	return (
		<div className='card'>
			<div className='time'>{props.title}</div>
			<img src={props.icon} alt="weather condition" />
			<h2 className='temp'>{props.temp ? props.temp : `${props.high} / ${props.low}`}</h2>
		</div>
	)
}

export default Card;