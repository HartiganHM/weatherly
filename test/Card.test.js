import React from 'react';
import { shallow, mount } from 'enzyme';
import Card from '../lib/Card';

const mockData = {
  time: '5:00 PM',
  src: 'http://icons.wxug.com/i/c/k/partlycloudy.gif',
  temp: '72°F',
  high: '212°F',
  low: '−459.67°F'
}

describe('Card Tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Card />)
  })

  it('Card should exist', () => {
    expect(wrapper).toBeDefined()
  })

  it('Should have a className card', () => {
    const card = shallow( <Card /> );
    expect( card.find('.card').length ).toEqual(1);
  })

  it('Should have a configurable time div', () => {
    const card = shallow( <Card time={mockData.time} />);
    expect( card.find('.time').length ).toEqual(1);
    expect('5:00 PM').toEqual(mockData.time);
  })
  
  it('Should have a configurable temp div', () => {
    const card = shallow( <Card temp={mockData.temp} />);
    expect( card.find('.temp').length ).toEqual(1);
    expect('72°F').toEqual(mockData.temp);
  })

  it('Should have a configurable temp div with high and low', () => {
    const card = shallow( <Card high={mockData.high} low={mockData.low} />);
    expect( card.find('.temp').length ).toEqual(1);      
    expect("212°F").toEqual(mockData.high);
    expect("−459.67°F").toEqual(mockData.low);
  })
})