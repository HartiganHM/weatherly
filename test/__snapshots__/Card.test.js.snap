// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Tests Should have a className card 1`] = `[Function]`;

exports[`Card Tests Should have a configurable src img 1`] = `[Function]`;

exports[`Card Tests Should have a configurable temp div 1`] = `[Function]`;

exports[`Card Tests Should have a configurable temp div with high and low 1`] = `[Function]`;

exports[`Card Tests Should have a configurable time div 1`] = `[Function]`;
